### PR TITLE
Release 0.4.1

### LINKED ISSUE #80: QDLDL types in qdldl_interface
The `qdldl_interface.[ch]` implementation is mixing types in the `bwork` working array.   In the header it is configured as a c_int and allocated as an array of c_ints, but then cast to QDLDL_bool when calling the ldl code.

It would be better to include `qdldl_types.h` directly in `qdldl_interface.h` and then use QDLDL specific types for all working arrays.   I have implemented something along these lines in the `qdldl_allocs` branch, but can't get it to build in cmake due to path problems for the headers. 

It might be safest to explicitly check that QDLDL_float and c_float match, as well as for the ints.   A semi-reliable way of doing this would just be`assert(sizeof(c_float) == sizeof(QDLDL_float))`, although there should be a nicer method than that.

### LINKED ISSUE #86: Should we expect OSQP_NON_CVX on all non-convex problems?
OSQP is happy to minimize -x^2 and return 0 as the optimal solution. Is this expected or should it detect non-convexity?
```julia
using OSQP
m = OSQP.Model()
OSQP.setup!(m, P=spdiagm([-1.0]))
OSQP.warm_start_x!(m, [2.0])
results = OSQP.solve!(m)
@show results.info.status
```

### PR DESCRIPTION
**Pull request by bstellato**
Update for release 0.4.1.

* Updated QDLDL to version 0.1.3.
* Fixes #86 
* Fixes #80

### CODE REVIEW

File: lin_sys/direct/qdldl/qdldl_interface.c [hunk 1]
```diff
@@ -47,7 +47,7 @@ void free_linsys_solver_qdldl(qdldl_solver *s) {
  * @param  p Private workspace
  * @return   exitstatus (0 is good)
  */
-static c_int LDL_factor(csc *A,  qdldl_solver * p){
+static c_int LDL_factor(csc *A,  qdldl_solver * p, c_int nvar){
```
Comment (bstellato) [inline lin_sys/direct/qdldl/qdldl_interface.c]: Can you please change the docs above to match the new function call?

**Comment by ahoarau** [comment]
Any way you could include qdldl directly in the project rather than a git submodule ? Otherwise the zip files in release are not valid ..

**Comment by bstellato** [comment]
@ahoarau @traversaro what if I automatically create a `.tar.gz` file with travis-ci every time we make a release? Would it work for you? It is not going to be in github but it should do the job. Also @traversaro sorry for my delay for the other issue you raised. I will get to it as soon as I can.

File: include/ctrlc.h [hunk 2]
```diff
@@ -34,18 +34,18 @@ bool utSetInterruptEnabled(bool);
 /**
  * Start listner for ctrl-c interrupts
  */
-void startInterruptListener(void);
```
Comment (lianglia-apollo) [inline include/ctrlc.h]: Can you add a namespace for the .h/.cc files?

Comment (bstellato) [reply include/ctrlc.h]: OSQP is written in C not C++. As far as I know, C does not have namespaces. Are you having any issues with conflicting function names? Please file a separate issue for that.

**Comment by traversaro** [comment]
> @ahoarau @traversaro what if I automatically create a .tar.gz file with travis-ci every time we make a release? Would it work for you? 

I never tried, but I guess once you generate a tarball in Travis it should be possible to add it to GitHub releases, see https://docs.travis-ci.com/user/deployment/releases/ .

> Also @traversaro sorry for my delay for the other issue you raised. I will get to it as soon as I can.

Don't worry, those are the kind of preemptive issues I typically raise in successful new projects, to avoid dealing with those tricky integration issues once they arise. Fixing them later, for example once a library has been package in debian, is typically extremly time consuming, but they are not causing us any problem at the moment. Thanks for the development and maintenance effort in OSQP! : )

**Comment by bstellato** [comment]
Commit 22b9ea0c1350957 makes sure the sources are deployed on bintray.com every time we tag a release. I decided to use that space because it is the same one where we upload the precompiled shared libraries for all the platforms.

**Comment by bstellato** [comment]
Done!

**Pull request merged**